packages/frontend/src/app/components/store/StoreProfile.tsx

```diff
@@ -67,31 +68,51 @@ const StoreProfile = ({ close }: { close: () => void }) => {
       stateManager.manifest.on("update", onUpdateEvent);
     };
   }, []);
-
   const copyToClipboard = () => {
     navigator.clipboard.writeText(shopId!);
   };
   const updateStoreInfo = async () => {
-    if (!baseAddr.length) {
+    if (!baseAddr) {
       setValidationError("Missing base currency address for store.");
     } else if (!baseChainId) {
       setValidationError("Missing base currency chainId for store.");
     } else if (!isValidHex(baseAddr)) {
       setValidationError("Base token address must be a valid hex value");
     } else {
       try {
-        const manifest: Partial<ShopManifest> = {
-          name: storeName,
-          setBaseCurrency: {
-            tokenAddr: baseAddr as TokenAddr,
+        const manifest: Partial<UpdateShopManifest> = {
+          setPricingCurrency: {
+            address: baseAddr as TokenAddr,
             chainId: baseChainId,
           },
         };
-        if (avatar) {
-          const path = await relayClient!.uploadBlob(avatar as FormData);
-          manifest["profilePictureUrl"] = path.url;
-        }
         await stateManager!.manifest.update(manifest);
+
+        //If avatar or store name changed, setShopMetadataURI.
+        if (avatar || storeName !== shopDetails.name) {
+          const metadata = {
+            name: storeName,
+            //If new avatar was uploaded, upload the image, otherwise use previous image.
+            image: avatar
+              ? (await relayClient!.uploadBlob(avatar as FormData)).url
+              : shopDetails.profilePictureUrl,
+          };
+          const jsn = JSON.stringify(metadata);
+          const blob = new Blob([jsn], { type: "application/json" });
+          const file = new File([blob], "file.json");
+          const formData = new FormData();
+          formData.append("file", file);
+
+          relayClient!.uploadBlob(formData).then(({ url }) => {
+            if (clientWallet && shopId) {
```

Comment:
im not sure i understand why we are doing this check here. Shouldn't we always have the clientWallet?